lib/events/event_router.rb

```diff
@@ -3,13 +3,14 @@ class EventRouter
     def self.dispatch(event_type, payload, headers = nil)
       case event_type
       when "Catalog.upload"
-        # TODO add service later
+        PersisterTaskService.new(payload).process
       when "Source.create"
         SourceCreateTaskService.new(payload).process
       when "Source.availability_check"
-        # TODO add service later
+        task = CheckAvailabilityTaskService.new(payload).process.task
+        task.dispatch
```

Comment:
@hsong-rh We also need to handle Source.delete event and delete the Source from the Catalog Inventory API database if the source has been deleted. The delete happens in the Sources API UI and we get a message that the Source has been deleted.

Reply:
@hsong-rh We need a begin/rescue block here to prevent our threads from ending. If we encounter something bad we want to be able to catch that exception and let the thread continue